### PR TITLE
chore/extended supported python and django versions

### PR DESCRIPTION
**Pull request by gecBurton**
I would like to extend the support for python to 3.11 and django to 4.2 so that we can use `django-gov-notify` in production

### CODE REVIEW

**Comment by nimasmi** [comment]
Thanks for this PR!

As per https://www.djangoproject.com/download/#supported-versions, the oldest currently-supported LTS release of Django is 3.2, and the oldest non-LTS release is 4.1. I would be happy to change the supported versions to just: "3.2", "4.1", "4.2".

**Comment by gecBurton** [comment]
> Thanks for this PR!
> 
> As per https://www.djangoproject.com/download/#supported-versions, the oldest currently-supported LTS release of Django is 3.2, and the oldest non-LTS release is 4.1. I would be happy to change the supported versions to just: "3.2", "4.1", "4.2".

that's suits us just fine, thanks for looking into this so quickly!

File: .github/workflows/CI-tests.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
     needs: [build, lint]
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
         django-version: ["3.2", "4.1", "4.2"]
```
Comment (nimasmi) [inline .github/workflows/CI-tests.yml]: According to https://docs.djangoproject.com/en/dev/faq/install/#what-python-version-can-i-use-with-django, the combination Django 3.2, Python 3.11 is not supported, 

Django version| Python versions
---|---
3.2|3.6, 3.7, 3.8, 3.9, 3.10 (added in 3.2.9)
4.0|3.8, 3.9, 3.10
4.1|3.8, 3.9, 3.10, 3.11 (added in 4.1.3)
4.2|3.8, 3.9, 3.10, 3.11
5.0|3.10, 3.11, 3.12
5.1|3.10, 3.11, 3.12

While this package is small, I don't really want to suggest things that may be misleading.

```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11"]
        django-version: ["3.2.9", "4.1.3", "4.2"]
        exclude:
            - django-version: "3.2"
              python-version: "3.11"
```

Comment (gecBurton) [reply .github/workflows/CI-tests.yml]: thank you for pointing this out, I have addressed it here [cc32b66](https://github.com/nimasmi/django-gov-notify/pull/8/commits/cc32b665d0dfc93de40ca7129168cbd0546b814c)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,19 +1,19 @@
 repos:
-  - repo: https://github.com/pre-commit/mirrors-isort
```
Comment (gecBurton) [inline .pre-commit-config.yaml:2]: no longer exists

File: django_gov_notify/message.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 
 class NotifyEmailMessage(EmailMessage):
-    """An email message that knows how to work with GOV.UK Notify templates. """
```
Comment (gecBurton) [inline django_gov_notify/message.py:9]: a new version of black is responsible for this

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -36,7 +36,7 @@ repos:
     hooks:
       - id: prettier
   - repo: https://github.com/awebdeveloper/pre-commit-stylelint
-    sha: c4c991cd38b0218735858716b09924f8b20e3812
```
Comment (gecBurton) [inline .pre-commit-config.yaml:39]: https://stackoverflow.com/questions/75216475/pre-commit-is-failing-with-missing-required-key-rev

**Comment by gecBurton** [comment]
hi @nimasmi are there any other revisions you would like me to make to this PR?

**Comment by nimasmi** [comment]
Thanks again @gecBurton. I have merged this via !10, and published 0.4.0 to PyPI: https://pypi.org/project/django-gov-notify/0.4.0/

**Comment by gecBurton** [comment]
> Thanks again @gecBurton. I have merged this via #10, and published 0.4.0 to PyPI: https://pypi.org/project/django-gov-notify/0.4.0/

thank you :)

**Pull request merged**